game/src/sandbox/mod.rs

```diff
@@ -678,6 +678,28 @@ impl State<App> for SandboxLoader {
                             self.stage = Some(LoadStage::GotScenario(scenario));
                             continue;
                         }
+                        gameplay::LoadScenario::Future(future) => {
+                            use map_gui::load::FutureLoader;
+                            return Transition::Push(FutureLoader::<App, Scenario>::new(
+                                ctx,
+                                Box::pin(future),
+                                "Loading Scenario",
```

Comment:
The major hurdle is any communication would need to be potentially cross thread. And App is pretty strongly locked into the main thread.

If there's some way to make a handle to Timer `Send`, I think this could get better... the immediate hurdle is that Timer.sink needs to be Send, which blows up lots of things.